src/player.js

```diff
@@ -286,6 +299,33 @@ class Player extends FakeEventTarget {
 
   //  <editor-fold desc="Playback Interface">
   /**
+   * The player readiness
+   * @public
+   * @returns {Promise<*>} - The ready promise
+   */
+  ready(): Promise<*> {
+    return this._readyPromise ? this._readyPromise : Promise.resolve();
```

Comment:
I think because we decided to link ready to the player constructor it means this is something related to player creation, which is done only once, so don't think we can reuse it, so no point to check this, as you make sure to create it on constructor so you know it is always there.
If on changing a media application wants to listen to any player related event they can do so by registering to events, so does our UI.